index.bs

```diff
@@ -4007,6 +4019,23 @@ In addition to that, truncating on byte boundaries alone causes a known issue th
 1. Ensure that any strings sent to authenticators are validly encoded.
 1. Handle the case where strings have been truncated resulting in an invalid encoding. For example, any partial code point at the end may be dropped or replaced with [U+FFFD](http://unicode.org/cldr/utility/character.jsp?a=FFFD).
 
+### Language and Direction Encoding ### {#sctn-strings-langdir}
+
+In order to be correctly displayed in context, the language and base direction of a string [may be required](https://www.w3.org/TR/string-meta/#why-is-this-important). Strings in this API may have to be written to fixed-function [=authenticators=] and then later read back and displayed on a different platform. Thus language and direction metadata is encoded in the string itself to ensure that it is transported atomically.
```

Comment:
I assumed that text libraries would require that such things came first in the string but, if not, suffixing acts better under truncation as you note. However, truncation becomes more complex because a language tag could be truncated into another valid language. For example, “ar-SA” into “ar”. That might be even worse than not knowing the language.

Thus, in order that the consumer can know whether the language has been truncated or not I have made either a directionality tag or U+E007F mandatory since it then acts as a terminator for the language tag. I've noted that a directionality tag should only be used if necessary to produce the correct result.

Reply:
@aphillips Are you ok with Adam's response ?

Reply:
@aphillips This is holding us from submitting for CR, so we would like to get this closed, please review